developer_guides/firmware/component-tutorial/tut-i-basic-fw-code.rst

```diff
@@ -0,0 +1,388 @@
+.. _developer_guides_tut-i:
+
+Part I - Adding a Component Code to FW
+######################################
+
+This lesson shows you how to add a component code to the FW source tree, with a
+minimal Component API implementation and simple copying function inside. It
+will also demonstrate how to register the component driver in the FW
+infrastructure so that the FW will be able to respond to the *new component*
+request sent by the driver and instantiate it.
+
+The amplifier will be based on a processing component class (aka effect).
+
+Adding Basic Component Code
+***************************
+
+New Component Type
+==================
+
+First, define a new component type in *src/include/ipc/topology.h*. It is a
+unique identifier used while declaring instances of the component as parts of
+the topology (more details on the required topology modifications will be
+provided in the next part of the tutorial, now we are just focusing on the FW
+source code).
```

Comment:
Is SW driver modification to handle comp ID going to be covered in the topology related section ?

Reply:
I'm going to create a reference to the driver's documentation with some brief intro what is required inside the driver's code to recognize the new _effect_'s type.